lib/src/native/realm_core.dart

```diff
@@ -1123,10 +1165,10 @@ extension on Pointer<IntPtr> {
 }
 
 extension on Pointer<Void> {
-  T? toObject<T extends Object>() {
+  T? toObject<T extends Object>([bool isPersistent = false]) {
```

Comment:
I think an optional `bool` argument is a bit secret here.. Maybe make it named, or have to different methods for weak and persistent.

Reply:
I was thinking of that myself. Still a single named argument seems too verbose and also this is not a public API but one for internal use. I think we should not be using named arguments by default. 
A different method will be better but extensions in the same file can not have private members as there aren't any so the same implementation would be repeated twice.

Reply:
I am a big proponent of using named arguments for bools.